src/components/Header/Header.tsx

```diff
@@ -0,0 +1,37 @@
+import * as React from 'react'
+import './Header.scss'
+
+interface Props {
+
+}
+
+const Header: React.FC<Props> = (props) => {
+  return (
+    <div className="header">
+      <div className="header_cover">
+        <div className="header_cover--black"></div>
+      </div>
+      <p className="header_logo">
+        <span>netflix</span>roulette
+      </p>
+
+      <button className="button button--grey header_addMovieButton">+ add movie</button>
+
+      <div className="header_searchWrapper">
+        <h2 className="header_title">Find your movie</h2>
+        <div className="header_searchWrapper_inputContainer">
+          <input
+            className="header_input input input--darkGrey"
+            type="text"
+            placeholder="What do you want to watch?" />
+
+          <button className="button button--magenta header_searchButton">
+            Search
+          </button>
+          </div>
+      </div>
+    </div>
+  )
+}
+
+export default Header
```

Comment:
A general idea about exports (taking your Header component as example):

- `interface HeaderProps`: Typedef for props, not need to be exported, as this is not supposed to be reusable, it is internal. Generally, we put a suffix `Props` after the component name, when defining the related props type.
- `export const Header`: It is good to export the "basic", unwrapped component at it's purest. Usually, we write unit test for this part only.
- `export default someOtherHoc(connect(Header))`: This is the final form of the component, used application-wide. This might be the place, to apply decorators, HOCs, and other additive functionality. For testing the base component, it is easier to only mock what is being injected by these wrappers, this way their behaviour is not tied to component during testing.

Reply:
Could you show an example or send some resource?

Reply:
If not a concrete resource, let me show it in a somewhat more meaningful context, like how I would define a component and related types using TS:

**components/header/header.tsx**

```typescript

/// imports as usual

// Note this is not exported - prop types are being local to a component, should not be reused elsewhere in this form
type HeaderProps = {
  lorem: number;
  ipsum?: string;
}

// I. Exporting as named
//    This is cool, because then this can be imported as `import  { Header } from 'components/header';`
// II. Note the default value being assigned to props.ipsum
// III. One should avoid using the React.FC type nowadays, only provide a type for props
// IV. In a unit test, you should only test this named export, as this contains the pure, basic component
export const Header = ({ lorem, ipsum = 10 }: HeaderProps ) => { ... }

// Imagine you had to wrap that Header with some I18n provider:
export default withIntl(Header);
```

**components/header/index.tsx**

```typescript
// If you had a decorated/wrapped component defined, you should export is like this
export { default as Header } from './header';

// Else you should just further export
export { Header } from './header';

// This is cool, because should you need any refactor/wrapping later, you would keep the
// public interface consistent, the following being still relevant:
// `import { Header } from 'components/header';`
```

**components/header/header.test.tsx**

```typescript
import { Header } from './header';

// Unit tests
describe('Header', () => {
  ...
});
```